neptune/new/handler.py

```diff
@@ -435,23 +414,30 @@ def download_last(self, destination: str = None) -> None:
         .. _download_last docs page:
            https://docs.neptune.ai/api-reference/field-types#download_last
         """
-        raise NeptuneException("Should be never called.")
+        return self._pass_call_to_attr(function_name="remove", destination=destination)
 
     def fetch_hash(self) -> str:
         """Fetches the hash of an artifact.
 
         You may also want to check `fetch_hash docs page`_.
            https://docs.neptune.ai/api-reference/field-types#fetch_hash
         """
-        raise NeptuneException("Should be never called.")
+        return self._pass_call_to_attr(function_name="fetch_hash")
 
     def fetch_files_list(self) -> List[ArtifactFileData]:
         """Fetches the list of files in an artifact and their metadata.
 
         You may also want to check `fetch_files_list docs page`_.
            https://docs.neptune.ai/api-reference/field-types#fetch_files_list
         """
-        raise NeptuneException("Should be never called.")
+        return self._pass_call_to_attr(function_name="fetch_files_list")
+
+    def _pass_call_to_attr(self, function_name, **kwargs):
+        attr = self._run.get_attribute(self._path)
+        if attr:
+            return getattr(attr, function_name)(**kwargs)
+        else:
+            raise AttributeError(f"No such method '{function_name}'.")
```

Comment:
I was thinking that we could add some information, that there's no such method, because field is not "initialized" yet, but:
1. it's quite due to our internal implementation, and probably this should be not user facing info
2. I wanted to keep behaviour compliant with the previous one